Add mon_getmap

cholcombe973 opened the pull request:
Adding a new function to get the latest monitor map.  This function required me to revisit my assumption that all things returned from ceph are String's so I changed the ceph_mon_command_without_data function.

cholcombe973 committed:
Add mon_getmap

Adding a new function to get the latest monitor map.  This
function required me to revisit my assumption that all things
returned from ceph are String's so I changed
the ceph_mon_command_without_data function.

ChrisMacNaughton commented on src/ceph.rs:
```diff
@@ -1891,7 +1891,7 @@ impl Rados {
         // back must be
         // freed by call `rados_buffer_free`
-        let mut str_outbuf: Option<String> = None;
+        let mut str_outbuf: Option<Vec<u8>> = None;
```
If we're changing the type, probably worth renaming the variable

ChrisMacNaughton commented on src/cmd.rs:
```diff
@@ -563,6 +563,7 @@ pub fn cluster_health(cluster_handle: &Rados) -> RadosResult<ClusterHealth> {
     });
     let result = cluster_handle.ceph_mon_command_without_data(&cmd)?;
     if let Some(return_data) = result.0 {
+        let return_data = String::from_utf8_lossy(&return_data);
```
Given that we are returning a Result anyways, would it be worth changing to `String::from_utf8(&return_data)?;`

ChrisMacNaughton commented on src/cmd.rs:
```diff
@@ -741,6 +746,24 @@ pub fn mon_dump(cluster_handle: &Rados) -> RadosResult<MonDump> {
+    return Err(RadosError::Error(format!(
+        "Unable to parse mon getmap output: {:?}",
+        result.1,
+    )));
```
why not do this dance as a `match`?

cholcombe973 committed:
Fix ceph_mon_command_without_data return type

After doing some digging through the ceph source code I found
out that just a RadosResult<Vec<u8>> should be returned instead
of the tuple that was being returned before.  It appears that
ceph only sets outs in situations where it knows more about the
error code.

cholcombe973 committed:
Delete more unnecessary code

ChrisMacNaughton commented:
I'd almost like to see the mon_getmap function added in a separate PR at this point :-P
